### ISSUE TITLE
Can the embeddings be used for your purposes?

### ISSUE THREAD

**Issue by VCasecnikovs** [getcursor/cursor#759]
I understand that embeddings are a lossy distillation of my code. As I understand, it is saved on your side.
Can you somehow use and analyze those embeddings?

**Comment by truell20** [comment]
I don't think so! (each embedding is a 1536 dimension vector that destroys a bunch of information)

Would you prefer we offer a completely local vector store? We avoided that because we assumed it would be a resource hog for large codebases and would be a bit tricky to deploy

**Comment by valh1996** [comment]
@truell20 The ideal would be to offer a choice. This is where you'll make the difference, because Cursor is a great product at the heart of the current AI trend. So if you also offer security and confidence for sensitive codes, you'll score a lot of points.

**Comment by mbtre** [comment]
I think it would be great to have that option. As embeddings get better, more and more information will be stored.

![image](https://github.com/getcursor/cursor/assets/128634931/e7d9d37e-baa5-43f8-a5d1-9925551ddaaf)
https://twitter.com/thesephist/status/1698095739899974031?s=20

The notion of someone being able to derive code from embeddings in future scenarios might not be as far-fetched as it seems.

**Comment by ybdesire** [comment]
+1 for completely local vector store

**Comment by madr3z** [comment]
+1 for local (or on premise) vector store. This is a red line for many businesses.
